browser/field-editor.js

```diff
@@ -68,69 +66,79 @@ function getField(elem, idx) {
 }
 
 function appendEditor(idx, elem) {
-    let $elem = $(elem)
-    let toEdit = getField($elem)
+  let $elem = $(elem)
+  let toEdit = getField($elem)
+  let text = $elem.attr("editText") || $elem.text().trim()
 
-    if (toEdit === null) {
-        return
-    }
-
-    let editClass = 'do-edit-' + toEdit + idx
+  if (toEdit === null) {
+    return
+  }
+  if (toEdit !== 'description' && toEdit !== 'title' && text === '') {
+    return
+  }
+  
+  let editClass = 'do-edit-' + toEdit + idx
 
-    let editLink = document.createElement('a')
-    editLink.setAttribute('style', 'margin-left: 10px')
+  let editLink = document.createElement('a')
+  editLink.setAttribute('style', 'margin-left: 10px')
 
-    let editButton = document.createElement('span')
-    editButton.setAttribute('class', 'fa fa-pencil ' + editClass)
-    editLink.appendChild(editButton)
+  let editButton = document.createElement('span')
+  editButton.setAttribute('class', 'fa fa-pencil ' + editClass)
+  editLink.appendChild(editButton)
 
-    $elem.append(editLink)
+  $elem.append(editLink)
 
-    $("." + editClass).on('click', () => {
-      let text = $elem.attr("editText") || $elem.text().trim()
-      let $input = $("<input/>").val(text)
+  $("." + editClass).on('click', () => {
+    let $input = $("<input/>").val(text)
+    
+    $elem.replaceWith($input)
 
-      $elem.replaceWith($input)
+    $input.one('blur', () => save($input, $elem, toEdit, text)).focus()
+  })
+  if ((toEdit === 'description' || toEdit === 'title') && text !== '') {
+    appendRemover(idx, elem)
+  }
+}
 
-      $input.one('blur', () => save($input, $elem, toEdit, text)).focus()
+function appendRemover(idx, elem) {
+  let $elem = $(elem)
+  let toRemove = getField($elem)
+  let removeClass = 'do-remove-' + toRemove + '-' + idx
+  let text = $elem.attr("editText") || $elem.text().trim()
+  if (text==='') {
+    return
+  }
+  let removeLink = document.createElement('a')
+  removeLink.setAttribute('class', "remove-" + toRemove)
+  removeLink.setAttribute('style', 'margin-left: 2px')
+  
+  let removeButton = document.createElement('span')
+  removeButton.setAttribute('class', 'fa fa-trash ' + removeClass)
+  removeLink.appendChild(removeButton)
+
+  $elem.append(removeLink)
+
+  $('.' + removeClass).on('click', () => {
+    let $trashes = $(".remove-" + toRemove)
+    let text = $elem.attr("editText") || $elem.text()
+    
+    $trashes.remove()
+
+    removeSparql(text, toRemove, () => {
+      location.reload()
```

Comment:
Why reload the page instead of just updating the field?

Reply:
Pages were getting messed up for a variety of reasons.  This was the only way I could think to make sure everything got to the right state even when errors occur.